descwl_coadd/coadd.py

```diff
@@ -276,11 +452,256 @@ def make_coadd(
+    # will zip these with the exposures to warp and add
+    wcss = [coadd_wcs, coadd_wcs, coadd_wcs]
+    bboxes = [coadd_bbox, coadd_bbox, coadd_bbox]
+    warpers = [warper, warper, mfrac_warper]
+
+    # PSF will generally have NO_DATA in warp as we tend to use the same psf
+    # stamp size for input and output psf and just zero out wherever there is
+    # no data
+
+    verifys = [verify]*3
+
+    LOG.info('warping and adding exposures')
+
+    if isinstance(exp, DeferredDatasetHandle):
+        expobj = exp.get()
+    else:
+        expobj = exp
+
+    # TODO use exp.getId() when it arrives in weekly 44.  For now use an index
+    # 0::len(exps)
+    try:
+        exp_id = expobj.getId()
+    except AttributeError:
+        exp_id = -9999
+
+    bad_msk, maskfrac = get_bad_mask(expobj)
+
+    noise_exp, medvar = get_noise_exp(
+        exp=expobj, rng=rng, remove_poisson=remove_poisson,
+    )
+
+    exp_info = get_info_struct(1)
+    exp_info['exp_id'] = exp_id
+    exp_info['maskfrac'] = maskfrac
+    exp_info['weight'] = 1/medvar
+
+    try:
+        mfrac_exp = make_mfrac_exp(mfrac_msk=bad_msk, exp=expobj)
+
+        if 0 < maskfrac < 1:
+            # images modified internally
+            interp_nocheck(exp=expobj, noise_exp=noise_exp, bad_msk=bad_msk)
+
+        # we use this to check no edges made it into the coadd
+        add_boundary_bit(expobj)
+        add_boundary_bit(noise_exp)
+
+        # order must match wcss, bboxes, warpers, verifys
+        exps2add = [expobj, noise_exp, mfrac_exp]
+
+        # the verify checks boundary/NO_DATA Use an exception as an easy way to
+        # guarantee that if one fails to verify we fail
+        warps = _get_warps_for_exp(exps2add, wcss, bboxes, warpers, verifys)
+        warp, noise_warp, mfrac_warp = warps
+
+        # We don't have a way of persisting ``medvar`` along with the warps and
+        # no easy way of computing it during coaddition, since it is computed
+        # from the equivalent of ``calexp``s.
+        # They may not be meaningful either, given that it is computed over
+        # many pixels beyond the cell boundary. So we check that the variance
+        # plane for ``noise_warp`` is same as that of ``warp`` and hijack it to
+        # store ``medvar``.
+        # This is intended to be a temporary hack to work on simulations,
+        # since with real data, we will run the `ScaleZeroPointTask` to scale
+        # to ``warp`` (and ``noise_warp``s) to have absolute photometric
+        # calibration.
+        np.testing.assert_array_equal(noise_warp.variance.array, warp.variance.array)
+        noise_warp.variance.array[:, :] = medvar
+
+    except WarpBoundaryError as err:
+        LOG.info('%s', err)
+        exp_info['flags'] |= WARP_BOUNDARY
+        warp, noise_warp, mfrac_warp = [None] * 3
+
+    return warp, noise_warp, mfrac_warp, exp_info
+
+
+def warp_psf(psf, wcs, coadd_wcs, coadd_bbox, psf_dims,
+             var=1.0, warper=None, filter_label=None):
```

Comment:
If the possibility of passing in a `warper` that may be configured differently from the one passed to `warp_exposures` worries you, please don't. We will have a 'contract' in the pipeline that will check that the configuration used here (during the coaddition stage) is the same one that was used during the warping stage.